contracts/implementation/PoolCommitter.sol

```diff
@@ -445,6 +456,38 @@ contract PoolCommitter is IPoolCommitter, Initializable {
             }
             counter += 1;
         }
+
+        (uint256 shortBalance, uint256 longBalance) = pool.balances();
+
+        uint256 longTotalSupply = IERC20(tokens[LONG_INDEX]).totalSupply();
+        uint256 shortTotalSupply = IERC20(tokens[SHORT_INDEX]).totalSupply();
+
+        updateMintingFee(
+            PoolSwapLibrary.getPrice(longBalance, longTotalSupply),
+            PoolSwapLibrary.getPrice(shortBalance, shortTotalSupply)
+        );
+    }
+
+    function updateMintingFee(bytes16 longTokenPrice, bytes16 shortTokenPrice) private {
+        bytes16 multiple = PoolSwapLibrary.multiplyBytes(longTokenPrice, shortTokenPrice);
+        if (PoolSwapLibrary.compareDecimals(one, multiple) == -1) {
+            // longTokenPrice * shortTokenPrice > 1
+            if (PoolSwapLibrary.compareDecimals(mintingFee, changeInterval) == -1) {
+                // mintingFee < changeInterval. Prevent underflow by setting mintingFee to lowest possible value (0)
+                mintingFee = 0;
+            } else {
+                mintingFee = PoolSwapLibrary.subtractBytes(mintingFee, changeInterval);
```

Comment:
Is there a risk that `mintingFee - changeInterval` > `MAX_MINTING_FEE` or is that impossible?

Reply:
I would say that's impossible, because `mintingFee` is capped at `MAX_MINTING_FEE` in the case where it increases, so `mintingFee - changInterval` must be strictly `LESS` than `MAX_MINTING_FEE` if `changeInterval > 0` or LESS THAN OR EQUAL to `MAX_MINTING_FEE` if `changeInterval == 0`.

Reply:
Yeah this was my thinking too just wanted to double check.